bluetooth-permissions/src/iosMain/kotlin/BluetoothPermissionManager.kt

```diff
@@ -54,18 +56,36 @@ actual class DefaultBluetoothPermissionManager(
     coroutineScope: CoroutineScope
 ) : BasePermissionManager<BluetoothPermission>(BluetoothPermission, settings, coroutineScope) {
 
+    companion object {
+        private fun checkAuthorization(): IOSPermissionsHelper.AuthorizationStatus {
+            val version = IOSVersion.systemVersion
+            return when {
+                version >= IOSVersion(13) -> CBManager.authorization.toAuthorizationStatus()
+                else -> CBPeripheralManager.authorizationStatus().toPeripheralAuthorizationStatus()
+            }
+        }
+    }
+
+    private class Delegate(
+        val authorizationStatusFlow: FlowCollector<IOSPermissionsHelper.AuthorizationStatus>,
+        private val coroutineScope: CoroutineScope
+    ) : NSObject(), CBCentralManagerDelegateProtocol {
+        override fun centralManagerDidUpdateState(central: CBCentralManager) {
+            val authorizationStatusFlow = authorizationStatusFlow
+            coroutineScope.launch {
+                authorizationStatusFlow.emit(checkAuthorization())
+            }
+        }
+    }
     private val permissionsQueue = dispatch_queue_create("BluetoothPermissionsMonitor", null)
     private val centralManager = lazy {
         val options = mapOf<Any?, Any>(CBCentralManagerOptionShowPowerAlertKey to false)
         CBCentralManager(null, permissionsQueue, options)
     }
 
-    private val delegate = object : NSObject(), CBCentralManagerDelegateProtocol {
+    private val permissionHandler = AuthorizationStatusHandler(eventChannel, logTag, logger)
 
-        override fun centralManagerDidUpdateState(central: CBCentralManager) {
-            handleAuthorizationStatus(checkAuthorization())
-        }
-    }
+    private val delegate = Delegate(permissionHandler, coroutineScope)
```

Comment:
I _think_ this is leaking the outer class (`this@[...].permissionHandler`) for freezing (could check with an 'ensureNeverFrozen()` on `init`). 

I guess it's not causing any problems right now, but the design seems 95% there to avoid this, and it looks like it is done elsewhere.